RIOT-OS/RIOT
# tools/elf2uf2: addition of new PROGRAMMER for RPi-pico

### Contribution description

Currently, RPi-pico (at least **RP2-B1** series) cannot be successfully flashed using **_uf2conv_** programmer. 

Suggested in the online documentation for RPi-pico board command:
```
make BOARD=rpi-pico PROGRAMMER=uf2conv flash
```
... did not work. The program is compiled and copied to the UF2 removable media (flash disk), but the board is not programmed. More details are described in the Issue [#15822](https://github.com/RIOT-OS/RIOT/issues/15822#issuecomment-985595835).

This PR adds new PROGRAMMER **_elf2uf2_**, which allows flashing RPi-pico boards using UF2 removable media (flash disk). Provided changes add conversion tool from elf file to UF2 format (using **_elf2uf2_** tool from [pico-SDK](https://github.com/raspberrypi/pico-sdk/tree/master/tools/elf2uf2)) as well as needed changes to makefiles and addition of new ones if required.

### Testing procedure



- press BOOTSEL button on RPi-pico board,
- connect RPi-pico to USB port, and wait for UF2 removable flash,
- go to examples/blinky directory,
- use command:
```
make BOARD=rpi-pico PROGRAMMER=elf2uf2 flash
```
- The board should be programmed, automatically reset, and the LED on the RPi-pico board should blink.

### Issues/PRs references

Issue [#15822](https://github.com/RIOT-OS/RIOT/issues/15822#issuecomment-985595835).
